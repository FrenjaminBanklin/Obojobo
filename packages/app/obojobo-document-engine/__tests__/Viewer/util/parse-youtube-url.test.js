import ParseYoutubeUrl from '../../../src/scripts/oboeditor/util/parse-youtube-url'

describe('YouTube Parse Url', () => {
	beforeEach(() => {
		jest.restoreAllMocks()
		jest.resetAllMocks()
	})

	test('test ParseYoutubeUrl youTubeParseUrl method with valid url', () => {
		const validUrl = 'https://www.youtube.com/watch?v=krfcq5pF8u8'
		const videoInfo = ParseYoutubeUrl.youTubeParseUrl(validUrl)

		expect(videoInfo).toMatchSnapshot()
	})

	test('test ParseYoutubeUrl youTubeParseUrl method with a different valid url', () => {
		const validUrl = 'https://www.youtube.com/watch?v=krfcq5pF8u8?t=5000'
		const videoInfo = ParseYoutubeUrl.youTubeParseUrl(validUrl)

		expect(videoInfo).toMatchSnapshot()
	})

	test('test ParseYoutubeUrl youTubeParseUrl method with valid embed code and start and end values', () => {
		const validUrl =
			'<iframe src=”https://www.youtube.com/embed/tpiyEe_CqB4?start=100&amp;end=300” allowfullscreen=”” width=”560″ height=”315″ frameborder=”0″>'
		const videoInfo = ParseYoutubeUrl.youTubeParseUrl(validUrl)

		expect(videoInfo).toMatchSnapshot()
	})

	test('test ParseYoutubeUrl youTubeParseUrl method with valid embed code and start and end values', () => {
		const validUrl =
			'<iframe src=”https://www.youtube.com/embed/tpiyEe_CqB4?start=50&end=345” allowfullscreen=”” width=”560″ height=”315″ frameborder=”0″>'
		const videoInfo = ParseYoutubeUrl.youTubeParseUrl(validUrl)

		expect(videoInfo).toMatchSnapshot()
	})

	test('test ParseYoutubeUrl youTubeParseUrl method with valid embed code', () => {
		const validUrl =
			'<iframe width="560" height="315" src="https://www.youtube.com/embed/ByH9LuSILxU?start=3" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>'
		const videoInfo = ParseYoutubeUrl.youTubeParseUrl(validUrl)

		expect(videoInfo).toMatchSnapshot()
	})

	test('test ParseYoutubeUrl youTubeParseUrl method with a valid embed url', () => {
		const validUrl = 'https://www.youtube.com/embed/Ho8niBS1nrU?start=539&end=541&version=3'
		const videoInfo = ParseYoutubeUrl.youTubeParseUrl(validUrl)

		expect(videoInfo).toMatchSnapshot()
	})

	test('test ParseYoutubeUrl youTubeParseUrl method with invalid url', () => {
		const validUrl = 'https://www.youtube.com/krfcq5pF8u8'
		const videoInfo = ParseYoutubeUrl.youTubeParseUrl(validUrl)

		expect(videoInfo).toMatchSnapshot()
	})

	test('test ParseYoutubeUrl youTubeParseUrl method with no url', () => {
		const videoInfo = ParseYoutubeUrl.youTubeParseUrl()

		expect(videoInfo).toMatchInlineSnapshot(`
		Object {
		  "endTime": false,
		  "startTime": false,
		  "videoId": false,
		}
	`)
	})
})
