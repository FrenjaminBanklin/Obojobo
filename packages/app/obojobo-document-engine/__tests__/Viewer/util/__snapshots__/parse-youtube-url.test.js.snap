// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YouTube Parse Url test ParseYoutubeUrl youTubeParseUrl method with a different valid url 1`] = `
Object {
  "endTime": 0,
  "startTime": "5000",
  "videoId": "krfcq5pF8u8",
}
`;

exports[`YouTube Parse Url test ParseYoutubeUrl youTubeParseUrl method with a valid embed url 1`] = `
Object {
  "endTime": 541,
  "startTime": "539",
  "videoId": "Ho8niBS1nrU",
}
`;

exports[`YouTube Parse Url test ParseYoutubeUrl youTubeParseUrl method with invalid url 1`] = `
Object {
  "endTime": false,
  "startTime": false,
  "videoId": false,
}
`;

exports[`YouTube Parse Url test ParseYoutubeUrl youTubeParseUrl method with valid embed code 1`] = `
Object {
  "endTime": 0,
  "startTime": "3",
  "videoId": "ByH9LuSILxU",
}
`;

exports[`YouTube Parse Url test ParseYoutubeUrl youTubeParseUrl method with valid embed code and start and end values 1`] = `
Object {
  "endTime": 300,
  "startTime": "100",
  "videoId": "tpiyEe_CqB4",
}
`;

exports[`YouTube Parse Url test ParseYoutubeUrl youTubeParseUrl method with valid embed code and start and end values 2`] = `
Object {
  "endTime": 345,
  "startTime": "50",
  "videoId": "tpiyEe_CqB4",
}
`;

exports[`YouTube Parse Url test ParseYoutubeUrl youTubeParseUrl method with valid url 1`] = `
Object {
  "endTime": 0,
  "startTime": 0,
  "videoId": "krfcq5pF8u8",
}
`;
